### ISSUE TITLE
T1.6 Members list

### ISSUE THREAD

**Issue by benjamincoleman** [IATI/IATI-Pattern-Library#123]
https://iatistandard.org/en/about/join-iati/members-assembly/

Want to make this: 

- Not a table
- sortable by year joined / alphabetical 
- filterable by type of organisation: 
    - Provider of development cooperation
    - Partner country
    - CSO / other

Use the pie chart image from 2018 report to show the spread of types of member!

There's a model for the content: 

[members-assembly.json.zip](https://github.com/IATI/IATI-Pattern-Library/files/3610161/members-assembly.json.zip)

--- 

https://marvelapp.com/4cfjg52/screen/61443823

Design: http://iati.surge.sh/page-templates/01-6-members-list

**Comment by benjamincoleman** [comment]
@BenDarby nice work. Some edits: 

- [ ] Can we add a white background to `.listing__aside .listing__media` to help them with image backgrounds and keep things consistent? 

- [ ] Active states for the filters please. This list is A-Z at the moment, so that one should be highlighted. 

- [ ] I'm not sure the grey bar really works as it shows at different heights depending on the content: 

![Screen Shot 2019-09-16 at 14 54 59](https://user-images.githubusercontent.com/64877/64963973-090a3700-d892-11e9-9490-e768cd4e08b8.png)

**Comment by benjamincoleman** [comment]
@BenDarby some changes for this template: 

- [ ] Could we make the text of the constituency and year joined under the name of the member larger 

- [ ] On the list of constituencies to filter by, could you add “Private sector organisations” (as IATI has recently added this constituency to its membership) 

- [ ] On the list of constituencies to filter by, could you replace ‘CSO and others’ with “Civil society organisations” (as IATI has recently changed this constituency name)

**Comment by benjamincoleman** [comment]
UNDP change: 

> See same comment in Marvel: Propose to add photos of leadership and their titles in the same way we’ve done [on this page](https://iatistandard.org/en/about/governance/who-runs-iati/). For the “who runs IATI” page and also here.

**Comment by benjamincoleman** [comment]
@BenDarby can you update this template adding 3 people above the list of members as seen here: https://marvelapp.com/4cfjg52/screen/61443823

You can get their images from here: https://iatistandard.org/en/about/governance/who-runs-iati/

**Comment by BenDarby** [comment]
@benjamincoleman I've updated the PL template to match the sketch.

**Comment by urlsangel** [comment]
@BenDarby @benjamincoleman 

http://iati.surge.sh/page-templates/01-6-members-list

Couple of questions:

Should the image/logo for each member be required, or optional with a fallback? We'll need a fallback image adding to the PL if the latter.

The PL list only goes up to the letter C - I'm guessing that means there's a whole lot more members. Are we paginating in that case, or is it going to be a really long page?

**Comment by urlsangel** [comment]
@benjamincoleman 

### Member filters and tags/metadata

The PL template has these filter terms:
- Civil society organisations
- Partner country
- Private sector organisations
- Provider of development cooperation

However, the data associated with individual members is in the form of:
- CSOs and others
- Providers of development cooperation
- Partner countries

These aren't the same text strings (and aren't a simple pluralised version either), so:

Are they the same thing? If so, is there a reason why they have different labels for filter terms and the individual items?

We can have a model that supports different text strings for filters vs items, but I'm not sure who benefits - as a user I might feel confused that the thing I'm filtering on isn't reflected in the data (`Civil society organisations` vs `CSOs and others` for example).

Finally, can a member be under more than one category?

**Comment by benjamincoleman** [comment]
Ok, 

1. the difference in naming is a content change, CSO = Civil Society Organisation 

2. members will only be of one type.

**Comment by benjamincoleman** [comment]
@BenDarby some updates needed on this template:

1. [ ] Can we make the logos links please
2. [ ] Need a fallback logo
3. [ ] Allow the titles to not be linked too - same for images as something are not linked.

**Comment by BenDarby** [comment]
@benjamincoleman when you say

> same for images as something are not linked 

do you mean the logos, ie have some linked and some not linked?

**Comment by benjamincoleman** [comment]
@BenDarby for some of the organisations there is no website, so there wont be any links on the title or image. 

When the title has no link, the line-height / margin is slightly different so we'll need to sort that out. It would probably be a good idea for titles with links to be a different colour too. 

You can see examples of this on the current members page, look for "Association of Freelance Journalists" : https://iatistandard.org/en/about/join-iati/members-assembly/

**Comment by BenDarby** [comment]
@benjamincoleman I've updated this template. We now have a variation where some are linked and some are not, there was border-bottom of 2px on linked headings before so I added a bottom margin of 2px on non-linked headings to address the difference as well as giving the linked headings the hover state as the standard style. I've used the IATI emblem for the fallback logo as it's not too specific, let me know if that's okay.
http://iati.surge.sh/page-templates/01-6-members-list
